common/script/ops/addWebhook.js

```diff
@@ -1,15 +1,22 @@
 import refPush from '../libs/refPush';
+import validator from 'validator';
+import i18n from '../i18n';
+import {
+  NotFound,
+  BadRequest,
+} from '../libs/errors';
 
-module.exports = function(user, req, cb) {
+module.exports = function(user, req) {
   var wh;
   wh = user.preferences.webhooks;
-  refPush(wh, {
+
+  if(!validator.isURL(req.body.url)) throw new BadRequest(i18n.t('invalidUrl', req.language));
```

Comment:
to make sure the server doesn't crash if req.body is missing I'd use `_.get()` (https://lodash.com/docs#get) to get the value from `req.body`